staging/src/k8s.io/legacy-cloud-providers/azure/azure_standard.go

```diff
@@ -403,7 +403,7 @@ func (as *availabilitySet) GetInstanceIDByNodeName(name string) (string, error)
 
 // GetPowerStatusByNodeName returns the power state of the specified node.
 func (as *availabilitySet) GetPowerStatusByNodeName(name string) (powerState string, err error) {
-	vm, err := as.getVirtualMachine(types.NodeName(name))
+	vm, err := as.getVirtualMachine(types.NodeName(name), cachedData)
```

Comment:
this should be unsafe. Instance id does not change.

Reply:
since this call is to get power status for the node, do we want to allow unsafe read?